core/quality/mysterium_morqa.go

```diff
@@ -291,9 +291,10 @@ func (m *MysteriumMORQA) ProviderSessions(providerID string) []ProviderSession {
 
 // ProviderStatuses fetch provider connectivity statuses from quality oracle.
 func (m *MysteriumMORQA) ProviderStatuses(providerID string) (node.MonitoringAgentStatuses, error) {
-	request, err := m.newRequestJSON(http.MethodGet, fmt.Sprintf("providers/statuses?provider_id=%s", providerID), "")
+	id := identity.Identity{Address: providerID}
+
+	request, err := requests.NewSignedGetRequest(m.baseURL, fmt.Sprintf("providers/auth-statuses?provider_id=%s", providerID), m.signer(id))
```

Comment:
Just something interesting to mention, might be useful for the future: you should not be able to upgrade the API so that you would no longer need to send the `provider_id` field in URL as you can extract the signer from signature.

Reply:
Yes, you are right. In side effect of auth we can get it inside signature in context

Reply:
yep, let's just use the identity from the signature, it should not be possible to pass anything except it.